pages/slice/language-reference/slice-grammar.md

```diff
@@ -2,14 +2,572 @@
 title: Slice grammar
 ---
 
+This page describes the grammatical rules, usage, and restrictions of the Slice language.
+
+## Slice elements
+
+### Slice files
+
+The `SliceFile` grammar rule is the parser's entry point.
+Once a file's contents have been [preprocessed], the resulting text is matched against this rule, in its entirety.
+
+While the contents of a Slice file may reference things defined in other files, each file is parsed separately.
+There is no notion of 'including' the contents of one file in another, and errors in one file do not affect the parsing of others.
+
+A Slice file consists of an optional [module declaration][module-declaration], followed by any number of Slice definitions.
```

Comment:
An empty file is a valid Slice file.
In the case where you don't have any Slice definitions, you're allowed to omit the module too.
We actually mention this in the `module declaration` section:
> If a Slice file contains no Slice definitions, it is legal to omit the module declaration.
Otherwise, if a Slice file contains any Slice definitions, a module declaration is required,